sklearn/utils/fixes.py

```diff
@@ -335,3 +335,20 @@ def __call__(self, *args, **keywords):
             return self.func(*args, **kwargs)
 else:
     from functools import partial
+
+
+if np_version < (1, 6, 2):
+    # Allow bincount to accept empty arrays
+    # https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040
+    def bincount(x, weights=None, minlength=None):
+        x = np.asarray(x, dtype=np.intp)
+        if x.size > 0:
+            return np.bincount(x, weights, minlength)
+        else:
+            if minlength is None:
+                minlength = 0
+            minlength = np.asscalar(np.asarray(minlength, dtype=np.intp))
+            return np.zeros(minlength, dtype=np.intp)
```

Comment:
This was modeled after [this numpy fix](https://github.com/numpy/numpy/commit/40f0844846a9d7665616b142407a3d74cb65a040). The `minlength = np.asscalar(np.asarray(minlength, dtype=np.intp))` line can probably be improved. It was meant as a loose substitute for PyArray_PyIntAsIntp.